code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -933,12 +933,12 @@
 	var/hrefpart = "<a href='?src=[REF(src)];track=[html_encode(namepart)]'>"
 	var/jobpart = "Unknown"
 
-	if (iscarbon(speaker))
-		var/mob/living/carbon/S = speaker
-		if(S.job)
-			jobpart = "[S.job]"
-	else
-		jobpart = "Unknown"
+	if (ishuman(speaker))
+		var/mob/living/carbon/human/S = speaker
+		if(S.wear_id)
+			var/obj/item/card/id/I = S.wear_id.GetID()
```

Comment:
can't you just check if GetID returns something instead of both wear_id and I

Reply:
No, because `GetID()` returns an inner ID sometimes (ex: PDAs, wallets) that can very much be null, `wear_id` only referring to the slot.